src/lib.rs

```diff
@@ -53,6 +54,17 @@ impl SignedSqrt {
     }
 }
 
+impl Add<SignedSqrt> for SignedSqrt {
+    type Output = Self;
+    fn add(self, other: Self) -> Self::Output {
+        SignedSqrt(
+            Rational::from_f64(
+                self.signed_sq().to_f64().sqrt() + other.signed_sq().to_f64().sqrt()
```

Comment:
One of the key guarantees of this package (and `SignedSqrt` in particular) is the absence of round-off errors associated with floating-point numbers. Is it possible to implement 12j symbols without floating-point arithmetic?

Reply:
OK, I will try.